### ISSUE TITLE
Social Icons: Add a 'Phone' icon which takes a tel: link

### ISSUE THREAD

**Issue by jordesign** [WordPress/gutenberg#55718]
In addition to having the 'email' icon it would be great to have a 'Phone' icon as well - which could take a `tel:` link to automatically add a link to make phonecalls.

**Comment by jordesign** [comment]
Suggested in the Forums:
https://wordpress.org/support/topic/phone-in-social-links-block/?view=all

**Comment by t-hamano** [comment]
I also think the "Phone" icon is necessary. However, we need to create an icon, so I would like to add a Needs Design label.

By the way, there is a Phone icon in[ the dash icon](https://developer.wordpress.org/resource/dashicons/#phone). This dash icon is also used in the experimental form block's tel field.

![form-tel](https://github.com/WordPress/gutenberg/assets/54422211/e49abd00-e85c-4c03-99c6-d71ddc7d47a5)

**Comment by skorasaurus** [comment]
similar issue at https://github.com/WordPress/gutenberg/issues/32890; but since this one has a clearer title; I'll keep this one open.

**Comment by colinduwe** [comment]
Submitted a PR that adds the variation using the Dashicon. There are a couple things to consider:
1. The url isn't empty, it starts with 'tel:' as the url with isn't how any others work. 
2. The description, "Display an icon linking to a social media profile or site.", isn't entirely appropriate here. I'm not sure if we can change that for variations.
<img width="490" alt="Screenshot 2023-12-15 at 1 49 36 PM" src="https://github.com/WordPress/gutenberg/assets/2152870/8bfe9199-6de0-4feb-873f-b96436262505">
<img width="299" alt="Screenshot 2023-12-15 at 1 49 43 PM" src="https://github.com/WordPress/gutenberg/assets/2152870/6f23f3d1-b5f8-42d6-9b8a-96598184c9b5">

**Comment by JosVelasco** [comment]
Hi, we could update the description to "Display an icon linking to a social media profile, website, or phone number."

As for the URL using tel:, I like how the e-mail mailto: is added automatically.

**Comment by colinduwe** [comment]
1. I still don't think we have a method to alter the text in the block title card per variation. For example the mail variation still reads:

<img width="281" alt="Image" src="https://github.com/user-attachments/assets/0ab45bf4-58da-4ffa-b368-a29489b5c485" />

2. I like the mail variation solution. If there's someone willing to review my PR I'll update it.

**Comment by t-hamano** [comment]
> I still don't think we have a method to alter the text in the block title card per variation.

We can change the description like this:

```javascript
const variations = [
	// ...
	{
		name: 'mail',
		attributes: { service: 'mail' },
		title: 'Mail',
		keywords: [ 'email', 'e-mail' ],
		icon: MailIcon,
		description: __( 'Display an icon linking to a mail address.' ),
	},
	// ...
]:
```

**Comment by colinduwe** [comment]
OK. PR is updated. Anyone able to review and approve it? https://github.com/WordPress/gutenberg/pull/57125

**Comment by karmatosed** [comment]
I think we jumped ahead a little which is ok, but let's have a moment to consider what the icon could be first. Let's do that here and as the 'needs design' label is added that should be path, not jumping straight to PR. We can do that and then get into the PR.

Let's consider:

- What should it look like based on the existing icon set?
- Is this needed or additional? We don't have enough opininions yet to move to 'needs design' I would suggest and would like to also get that input.

I realise this might feel a step back, but we have to consider when adding assets like icons and this type of functionality.

**Comment by JosVelasco** [comment]
Hi @karmatosed,

I think the standard phone icon is good enough to start. It's clean and easy to understand. 

The smartphone icon could be cool to introduce as a variation, but it is not really needed for now, in my opinion.

Adding the phone option now that we have mail can help greatly for users who haven't defined their social networks yet. In my experience, most users have their phone and mail before social networks.

**Comment by colinduwe** [comment]
Sorry to have jumped the gun with my PR back in December 2023. Happy to incorporate any feedback. I do feel like the current pace is pretty slow. So maybe it would be OK to move ahead with the variation with the current icon and then we can discuss the design and release an improvement in future years.

**Comment by mtias** [comment]
It's a bit weird to have email and phone under social icons. Curious if they should be standalone blocks that can also be used inside the social icons block.

**Comment by JosVelasco** [comment]
They would feel too specific on their own, I think. 

An email or phone block that is only a link? I'd expect more functionality than that.

What about just giving them less relevance instead but still making them available?

**Comment by t-hamano** [comment]
Now, there is an official way to register custom variations on both the editor and front end. See https://github.com/WordPress/gutenberg/pull/70261

Additionally, a guideline about adding new social icons has been documented: https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/social-link/README.md#adding-new-social-icons

I am wondering whether we should close this isse as "won't fix", but since the telephone is not a social service, it is difficult to know how to make a decision based on the guidelines.

cc @Mamaduka @bph @priethor

**Comment by Mamaduka** [comment]
Don't have a strong opinion, but since the block already has services Email and WhatsApp, we might as well add Phone "service".

**Comment by priethor** [comment]
If we have Email, I see a strong case for adding Phone as well. Perhaps we shouldn't view it as social services, but rather as ways to contact.

**Comment by colinduwe** [comment]
Adding telephone is frequently requested so fits the final bullet point of the read me. And I’d argue really improves the UX of building a site since its such a common design pattern to have the business’s social, email, and phone as attractive icons in the header, footer, etc.

**Comment by bph** [comment]
I concur with @Mamaduka @priethor and @colinduwe Telephone number is still one of the basic contact information and should have a social icon link. @mtias maybe we should rename the social icon block to "My profiles block"

**Comment by JosVelasco** [comment]
Instead of renaming it, I prefer a name that includes more use cases, like "Contact & Social Icons block." This way, there is no need to learn a new name.

**Comment by JosVelasco** [comment]
Or maybe "Social & Contact Icons block."

**Comment by colinduwe** [comment]
Perhaps the renaming of the parent block discussion should be spun off into a separate issue so we can keep this issue moving ahead delivering this particular block variation?

**Comment by JosVelasco** [comment]
@karmatosed do you feel we have enough opinions to add 'needs design'? I think so!

**Comment by Mamaduka** [comment]
I don't think we should rename the block. That's a separate discussion unrelated to this request.

P.S. It has already been renamed once.

**Comment by t-hamano** [comment]
While there's no strong opinion on whether to add a phone icon, another challenge is how to distinguish phone numbers.

For the Email icon, we use a handy function called `is_email()`, which automatically adds the `mailto:` protocol as a prefix:

https://github.com/WordPress/gutenberg/blob/9b5c92503fc93e1495784a0917c3fa7c28e54dec/packages/block-library/src/social-link/index.php#L35-L41

We'd probably need to add the `tel:` protocol to the phone link as well, but there's no `is_tel()` function.

It would not be easy to distinguish between phone number standards, which vary greatly from country to country.

See https://github.com/WordPress/gutenberg/pull/57125#discussion_r2106066619

**Comment by mtias** [comment]
Yeah, sorry don't mean to derail. Social & Contact or something with Links could make sense. We should also ensure things show up for inserter searches.

**Comment by colinduwe** [comment]
@t-hamano i just responded to your PR comment. Confirming any given string is a valid telephone number does seem to be an unsolvable problem. But I don’t think we need to do that. The user can ensure they entered the right numbers. We can help them prepend “tel:” and call it a day.

**Comment by t-hamano** [comment]
> The user can ensure they entered the right numbers. We can help them prepend “tel:” and call it a day.

Does this mean that it doesn't check if the URL is in phone number format, but always prefixes the URL with `tel:` if the Phone social link block is used?

**Comment by colinduwe** [comment]
Correct. Like the email prefixes with mailto:, it seemed reasonable to prefix with tel:. My current PR also tries to remove invalid dialing characters
https://github.com/WordPress/gutenberg/blob/0b456c3b7cb33196276b394ebd7462ef2bb34326/packages/block-library/src/social-link/index.php#L39-L50

**Comment by t-hamano** [comment]
Things seem mostly positive about adding a phone icon, so let's move forward with #57125. However, we need a polished icon to ship that PR. Current screenshot:

<img width="588" height="212" alt="Image" src="https://github.com/user-attachments/assets/64ba3a0e-a1d2-407d-9251-9f8844802e49" />

**Comment by karmatosed** [comment]
I would like to consider a more modern icon if we have one than the one showing up. Do we have one that perhaps scales better at smaller sizes @t-hamano? My conecrn is that icon looks rather dated and hard to read at small sizes if there is a decision to go with a phone number. 

Let’s for now get ‘needs design’ on this and see what options we could have if everyone is set on this. I still have concerns with the need for an icon for all types of contact, but that’s a separate issue and perhaps personal preference.

**Comment by t-hamano** [comment]
The icon SVG proposed in #57125 is likely based on [the dashicon](https://developer.wordpress.org/resource/dashicons/#phone). Since the dashicon is not suitable for Gutenberg, we need to create a new icon.

**Comment by t-hamano** [comment]
I noticed that #71514 adds a telephone control with the `mobile` icon:

<img width="340" height="234" alt="Image" src="https://github.com/user-attachments/assets/887ce2f5-b2ed-4ef3-ae9b-eee5094276eb" />

If I think about it, the "phone receiver" icon is used as the phone icon on many smartphones, but the mobile icon might represent a phone in modern times. What do you think? Users who want to change this icon can use the `block_core_social_link_get_services` hook.

**Comment by jhmonroe** [comment]
@t-hamano interesting points about old phone receiver style vs mobile! I also perused your slack convo about this topic from earlier this year and looked thru issues/make wordpress to see what was out there. Just putting the following all in one place:

Regarding who designed all the existing icons, I found a few interesting threads that might help find nice icons, consistent icons, or people who consolidated what exists:

Here's the Figma Wordpress Components Library with mobile icon that looks similar to what you shared above: https://www.figma.com/design/CtTrCB3BYfNaieyAnFzUHr/WordPress-Design-System--Community-?node-id=15040-40302&t=IgJtLuT0AV1t3IIp-0

The most recent conversations (outside Slack) I could find were from 2020:
Including one by @jasmussen about Dashicons, Material icons, and custom icons:
On merging all icon sources into one: https://github.com/WordPress/gutenberg/issues/10502
Where they netted out: https://make.wordpress.org/design/2020/04/20/next-steps-for-dashicons/
On icon creation in general: https://github.com/WordPress/dashicons/issues/255#issuecomment-358948361

Agree with others that the Dashicons receiver icon is a bit too detailed/fussy with the double line compared to the simplicity of the other social icons like Mail.

The Material phone icon in the rounded variant is still a bit odd/angular, but perhaps as small nav sizes would work, and the mail icon is already quite similar to the Material Mail icon (perhaps someone tweaked it *a bit*): https://fonts.google.com/icons?selected=Material+Symbols+Rounded:phone_enabled:FILL@0;wght@400;GRAD@0;opsz@24&icon.query=phone&icon.style=Rounded&icon.size=24&icon.color=%231f1f1f

Here something called Primary G2 is mentioned but I can't find what those icons are: https://make.wordpress.org/design/?s=dashicons

Curious where they landed with all this icon library merging! :-)


> I noticed that [#71514](https://github.com/WordPress/gutenberg/pull/71514) adds a telephone control with the `mobile` icon:
> 
> <img alt="Image" width="340" height="234" src="https://private-user-images.githubusercontent.com/54422211/486355006-887ce2f5-b2ed-4ef3-ae9b-eee5094276eb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc4MTE4MzEsIm5iZiI6MTc1NzgxMTUzMSwicGF0aCI6Ii81NDQyMjIxMS80ODYzNTUwMDYtODg3Y2UyZjUtYjJlZC00ZWYzLWFlOWItZWVlNTA5NDI3NmViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTE0VDAwNTg1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4MDQ5ZDJmZGM5NDcwYTRjMzgxMWRlOWM0NWIxYmM3YmQ5ZGEwOTg5NThmZDE5ZTM5MmFkZTNiZjIzZWEwN2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UnGfoxZGqNGMkc1GuXS9WsJUYcaE-RyijqtWfRM3irA">
> If I think about it, the "phone receiver" icon is used as the phone icon on many smartphones, but the mobile icon might represent a phone in modern times. What do you think? Users who want to change this icon can use the `block_core_social_link_get_services` hook.

**Comment by jhmonroe** [comment]
Interesting point: "The telephone icon isn't outdated, it is the opposite. It's ingrained in culture now, just like the floppy disk icon to save a file. Icons can have semantic meaning, like the biohazard sign, even though it doesn't resemble anything "real"."

Source: https://x.com/usgraphics/status/1976716595117629494?s=46

**Comment by karmatosed** [comment]
@jhmonroe I would disagree with you and that link is very one culture. Ask different generations to describe a telephone they will not say the same thing.
